src/main.rs

```diff
@@ -188,11 +190,15 @@ async fn main() {
         tracing::error!("Failed to save playerlist: {:?}", e);
     }
 
+    // Friendslist
+    task::spawn(load_friends_list(settings.clone()));
```

Comment:
Anything that touches the global state (i.e. this call) should be dispatched after `State::initialize_state` otherwise there is a race condition that could cause a panic. This could also allow the called function to retrieve the player and API key from the state rather than having to clone it.

This does bring the fact that this is possible to my attention though, so maybe in the future it should be refactored to pass around `Arc<RwLock<State>>` instead of relying on a global state that needs to be initialized before doing anything else. I might make an issue for it.